src/include/miopen/mha/invoke_params.hpp

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/invoke_params.hpp>
+#include <miopen/tensor.hpp>
+#include "mha.hpp"
+
+namespace miopen {
+namespace mha {
+
+struct InvokeParams : public miopen::InvokeParams
+{
+    InvokeParams(const MHAInputDescsForward& inputDescsForward,
+                 const MHADataForward& dataForward,
+                 Data_t ws,
+                 std::size_t wsSize)
+        : mhaInputDescsForward(inputDescsForward),
+          mhaDataForward(dataForward),
+          workSpace(ws),
+          workSpaceSize(wsSize)
+    {
+    }
+
+    const MHAInputDescsForward& GetDescs() const { return mhaInputDescsForward; }
+    const MHADataForward& GetData() const { return mhaDataForward; }
+
+    std::size_t GetWorkspaceSize() const { return workSpaceSize; }
+    Data_t GetWorkspace() const { return workSpace; }
+
+private:
+    MHAInputDescsForward mhaInputDescsForward;
```

Comment:
Can you please remind me why do we need a copy of `MHAInputDescsForward`? It's size is not that small.

/cc @DrizztDoUrden

Reply:
It's better to save to not rely on an user of this class.

Reply:
@Vsevolod1983 @DrizztDoUrden let's discuss this topic offline.

Reply:
@atamazov 
Actually, pretty same reason why we don't use const references to Tensor Descriptors but use copy of Tensor Descriptors instead.
Was discussed here:
https://github.com/ROCm/MIOpen/pull/2671#discussion_r1466920140

Reply:
Could we merge this PR to unblock other PRs creation ?

Reply:
@Vsevolod1983 This comment cannot be considered resolved, but this is not a blocker to merging this PR.

Reply:
[future][performance] To be discussed offline.